projects/packages/forms/src/modules/file-field/view.js

```diff
@@ -236,16 +350,45 @@ store( NAMESPACE, {
 		},
 
 		/**
-		 * Remove a file from the context.
+		 * Remove a file from the context and cancel its upload if in progress.
 		 *
 		 * @param {Event} event - The event object.
+		 * @yield {Promise<string>} The upload token.
 		 */
-		removeFile: event => {
+		removeFile: function* ( event ) {
 			event.preventDefault();
+
+			const { ref } = getElement();
+			const field = ref.parentElement.parentElement.parentElement; // Needed to select the top most field.
+			clearInputError( field, { hasInsetLabel: isInlineForm( ref ) } );
```

Comment:
Looking at what the `clearInputError` function does, it seems to manipulate the HTML imperatively. Wouldn't it be better to do it declaratively using the Interactivity API, or is that a part of the HTML that doesn't use the Interactivity API yet?